eclipse/che-theia
# feat(recommendations): Provide a plug-in for recommendations

### What does this PR do?
Provide a plug-in for recommendations

1. Open a workspace without plug-ins: provide featured plug-ins and ask to reload workspace
2. Open a workspace containing a devfile with plug-ins: suggest it (but not prompting restart)

3. When opening file: Inform that some plug-ins may exist  (only if `extensions.openFileRecommendations` is set to true in devfile)


### Screenshot/screencast of this PR



### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/18274


### How to test this PR?
Starts a new workspace with this plug-in and a devfile without plug-ins but with java or go project.
--> should prompt to restart workspace and then have intellisense

Try to open some files like go or python: should suggest that existing plug-ins may be there (`extensions.openFileRecommendations` is set to true in devfile attributes)

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### Happy Path Channel

HAPPY_PATH_CHANNEL=next
